packages/components/src/core/Dropdown/index.test.tsx

```diff
@@ -40,4 +61,154 @@ describe("<Dropdown />", () => {
     fireEvent.click(DropdownElement);
     expect(screen.getAllByText("Loading")).not.toBeNull();
   });
+
+  it("Prop Test (search): Dropdown component search function correctly maintains searched-for items", () => {
+    const args = {
+      ...Test.args,
+      search: true,
+    };
+
+    render(<Test {...args} />);
+
+    // Open the dropdown
+    const DropdownElement = screen.getByTestId("dropdown");
+    fireEvent.click(DropdownElement);
+
+    // Select the search input by placeholder
+    const searchInput = screen.getByPlaceholderText("Search");
+    fireEvent.change(searchInput, { target: { value: "Status" } });
+
+    // We want to test that there ARE items with "Status: *" in the dropdown menu
+    // because we have searched for Status
+    const statusItem = screen.queryByText("Status: confirmed");
+    expect(statusItem).not.toBeNull();
+  });
+
+  it("Prop Test (search): Dropdown component search function correctly filters out Type items", () => {
+    const args = {
+      ...Test.args,
+      search: true,
+    };
+
+    render(<Test {...args} />);
+
+    // Open the dropdown
+    const DropdownElement = screen.getByTestId("dropdown");
+    fireEvent.click(DropdownElement);
+
+    // Select the search input by placeholder
+    const searchInput = screen.getByPlaceholderText("Search");
+    fireEvent.change(searchInput, { target: { value: "Status" } });
+
+    // We want to test that there are no items with "Type: *" in the dropdown menu
+    // because we have searched for Status
+    const typeBugItem = screen.queryByText("Type: bug");
+    expect(typeBugItem).toEqual(null);
+  });
+
+  it("Prop Test (Label): Dropdown component correctly renders with a new label", () => {
+    const args = {
+      ...Test.args,
+      label: "Test Label",
+    };
+
+    render(<Test {...args} />);
+    expect(screen.getAllByText("Test Label")).not.toBeNull();
+  });
+
+  it("Prop Test (closeOnBlur): Dropdown closes when button is clicked with closeOnBlue set to true", async () => {
+    const args = {
+      ...Test.args,
+      closeOnBlur: true,
+    };
+
+    render(<Test {...args} />);
+
+    // Open the dropdown
+    const DropdownElement = screen.getByTestId("dropdown");
+    userEvent.click(DropdownElement);
```

Comment:
Doe sit actually need to use `userEvent` here, or would `fireEvent` work now that it has `wait(500)` below?

Reply:
Actually, it doesn't work with fireEvent, and we kinda have to use userEvent in this case!

Reply:
Got it!